### PR TITLE
update istioctl pc workload command for latest ambient wds

### PR DESCRIPTION
**Pull request by GregHanson**
**Please provide a description of this PR:**

follow up to https://github.com/istio/istio/pull/44199

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ X ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ X ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by GregHanson** [comment]
example output after fix:
```
NAME                                          NAMESPACE    IP           NODE                                            WAYPOINT               PROTOCOL
event-exporter-gke-755c4b4d97-f2mjf           kube-system  10.124.1.5   gke-gihanson-ambient-default-pool-090bfae6-6dj4 None                   TCP
httpbin-797587ddc5-b74np                      default      10.124.0.48  gke-gihanson-ambient-default-pool-090bfae6-qrkz httpbin-istio-waypoint HBONE
httpbin-istio-waypoint-6d86d5bd9-9pn8v        default      10.124.2.115 gke-gihanson-ambient-default-pool-090bfae6-grql None                   TCP
istio-egressgateway-789d9959fd-r2zgt          istio-system 10.124.2.108 gke-gihanson-ambient-default-pool-090bfae6-grql None
...
...
```

File: istioctl/pkg/writer/ztunnel/configdump/workload.go [hunk 1]
```diff
@@ -141,18 +139,14 @@ func (c *ConfigWriter) retrieveSortedWorkloadSlice() (*ztunnelDump.ZtunnelDump,
 	return workloadDump, nil
 }
 
-func waypointName(wl *ztunnelDump.ZtunnelWorkload, prevAddresses map[string]string, workloads []*ztunnelDump.ZtunnelWorkload) string {
-	if len(wl.WaypointAddresses) == 0 {
+func waypointName(wl *ztunnelDump.ZtunnelWorkload, services map[string]*ztunnelDump.ZtunnelService) string {
+	if wl.Waypoint == nil {
```
Comment (hanxiaop) [inline istioctl/pkg/writer/ztunnel/configdump/workload.go:143]: Can `wl` be nil here? Should we avoid panic?

**Review by hanxiaop — COMMENTED** [review]
Generally LGTM

File: istioctl/pkg/writer/ztunnel/configdump/workload.go (hunk 1 above)
Comment (GregHanson) [reply istioctl/pkg/writer/ztunnel/configdump/workload.go:143]: we're cycling through a list of verified workloads in `istioctl/pkg/writer/ztunnel/configdump/workload.go:PrintWorkloadSummary` - so I don't think it's a concern unless the print summary function changes drastically

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**